scalyr_agent/util.py

```diff
@@ -2434,3 +2436,226 @@ def match_glob(pathname):
         # use the third party glob library to handle a recursive glob.
         result = glob2.glob(pathname)
     return result
+
+
+class ProcessWatchDog(threading.Thread):
+    """
+    A watchdog that creates a new thread and pools a particular process.
+    If the process does not exist anymore, the watchdog invokes a callback.
+    """
+
+    def __init__(self, pid, poll_interval):
```

Comment:
`pid` could mean the current process's id.  I would call this `pid_to_watch`.  Use that for the parameter and the class instance variable.  I don't love `pid_to_watch` so if you come up with something better, than that's fine as well.

Reply:
What about `target_pid`?

Reply:
Sure, that's fine.